### PR TITLE
remove deprecated fields from v1 types

### LINKED ISSUE #8298: Dedupe SecurityContext fields in PodSpec
Capabilities and Privileged are deprecated and should be removed from the v1 API.

cc @pmorie @pweil-

### PR DESCRIPTION
**Pull request by pweil-**
Resolves https://github.com/GoogleCloudPlatform/kubernetes/issues/8298

Removing the deprecated fields also allows us to remove the conversion override, defaulting, and defaulting tests that were used to validate mismatches on the deprecated fields, set defaults post-conversion, and do upward/downward merges pre-conversion.

Conversion Master @wojtek-t PTAL (there seemed to be some ordering changes by the generator that created some noise in the diffs as well since capabilities is no longer on the Container object)

/cc @pmorie @erictune @bgrant0607

### CODE REVIEW

**Comment by vmarmol** [comment]
Needs a rebase btw

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -640,12 +640,8 @@ type Container struct {
 	Lifecycle      *Lifecycle           `json:"lifecycle,omitempty" description:"actions that the management system should take in response to container lifecycle events; cannot be updated"`
 	// Optional: Defaults to /dev/termination-log
 	TerminationMessagePath string `json:"terminationMessagePath,omitempty" description:"path at which the file to which the container's termination message will be written is mounted into the container's filesystem; message written is intended to be brief final status, such as an assertion failure message; defaults to /dev/termination-log; cannot be updated"`
-	// Deprecated - see SecurityContext.  Optional: Default to false.
```
Comment (wojtek-t) [inline pkg/api/v1/types.go]: @bgrant0607 what is the policy of deprecation? Is it ok to remove it now?

Comment (bgrant0607) [inline pkg/api/v1/types.go]: v1 is still completely changeable. It hasn't been finalized yet and is not exposed by default.

**Comment by bgrant0607** [comment]
API change LGTM. Thanks!

File: pkg/api/v1/types.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/api/v1/types.go]: Thanks Brian! I will review the other changes later today.

**Comment by wojtek-t** [comment]
LGTM

**Pull request merged**